### ISSUE TITLE
Bot table

### ISSUE THREAD

**Issue by anbu-squad** [UnamSanctam/SilentCryptoMiner#2245]
I want to add a table of bots (devices) to the panel
and I need advice on how to properly organize filling out the table
Most likely you need to write a new code for sending requests in the body of the build?

I took part of the data from the miners table and added country codes
however, my status is displayed in the context of the miner's work
but the bot needs a different context

In general, I want to try to implement the following format

ID - IP - Country - Username - Computername - GPU - CPU - Status minig CPU - Status mining GPU

**Comment by anbu-squad** [comment]
What delay should be set in requests?

**Comment by UnamSanctam** [comment]
You mean like a completely different program? Since all the things you listed already get sent by the miners. If it's a different program then creating a new endpoint and table in the database would be how you'd do it.

**Comment by anbu-squad** [comment]
I want to add a loader for requests, where in the server response, if necessary, you can specify the URL
To organize the correct operation of the loader, I want to organize a table of devices for ease of searching for the bot by ID

and I can’t figure out how to properly organize a post request

Does each miner send post requests separately?

does this mean I need to add new code that will send new requests?

**Comment by UnamSanctam** [comment]
Yes the injected miners are independent of each other and each send their own requests, though they all share the same Unique Hash so you can always group them. If you want to create a table of the computers then you can simply group the miners into a single row depending on their unique hash and display that, it doesn't sound like there's a need for any new post requests.

**Comment by anbu-squad** [comment]
I understand that it will be easier to organize a separate table, this question was also interesting to me about the unique hash

But for the loader, so that I understand the logic correctly, I need to add sending post requests separately in the miner build itself?

**Comment by UnamSanctam** [comment]
If by loader you mean the program that installs itself (if enabled), injects the miners and then closes down then if you want that to send a request then yes you'll need to add something that sends requests inside that (such as cpp-httplib).

**Comment by anbu-squad** [comment]
I'm considering the option cpp-httplib + urlmon.lib

the build comes out that introduces miners and closes?

This means I can implement a third process (loader)
is my logic correct?

`SELECT ms_uqhash, COUNT(*) as count
FROM bots
GROUP BY ms_uqhash;
`

With this query I can group miners into devices in the bots table?

**Comment by UnamSanctam** [comment]
Yes you could implement a third process like the watchdog which runs alongside the miners and sends requests.

>With this query I can group miners into devices in the bots table?

That query would count the amount of devices yes (like how it's done on the statistics page), though that's all it would do.

**Comment by anbu-squad** [comment]
great, thanks, the tips helped me a lot
I've been studying the panel logic for several days now
and it can be difficult to formulate the right tasks for artificial intelligence

This means grouping by a unique hash and a conditionally new process for the loader

**Comment by anbu-squad** [comment]
API Endpoint URL

in the code itself are stored as an array?
and is there a logic for checking the server response?
if the server's response is not 200
does it mean it goes to the next element in the array?

**Comment by UnamSanctam** [comment]
Yes it's stored as an array, and then the URLs are checked in order and if the response status is not between 200 and 400 and not a valid JSON then it tries the next one in the array.
